File: cpp/sprint2/B/code.cpp
// ! Remove define before submitting !
#define local

#ifndef local
#include "solution.h"
#endif


#ifdef local

#include <string>

struct Node {  
    std::string value;  
    Node* next;  
    Node(const std::string &value, Node* next);  
};

#endif

void solution(Node* head) {
    // Your code
    // ヽ(´▽`)/
}

void test() {
    Node node3("node3", nullptr);
    Node node2("node2", &node3);
    Node node1("node1", &node2);
    Node node0("node0", &node1);
    solution(&node0);
    /*
    Output is:
    node0
    node1
    node2
    node3
    */
}

File: python/sprint2/D/code.py
# ! change LOCAL to False before submitting !
# set LOCAL to True for local testing

LOCAL = False

if LOCAL:
    class Node:
        def __init__(self, value, next_item=None):
            self.value = value
            self.next_item = next_item


def solution(node, elem):
    index = 0
    while True:
        if node == None:
            return -1
        if node.value == elem:
            return index
        node = node.next_item
        index += 1


def test():
    node3 = Node("node3", None)
    node2 = Node("node2", node3)
    node1 = Node("node1", node2)
    node0 = Node("node0", node1)
    idx = solution(node0, "node2")
    assert idx == 2


if __name__ == '__main__':
    test()
